Sources/PT.PM.Common/PathUtils.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace PT.PM.Common.Utils
+{
+    public static class PathUtils
+    {
+        public const int MaxDirLength = 248 - 1;
+        public const int MaxPathLength = 260 - 1;
+
+        public static string NormalizeFilePath(this string path) => NormalizePath(path, false);
+
+        public static string NormalizeDirPath(this string path, bool force = false) => NormalizePath(path, true, force);
+
+        private static string NormalizePath(this string path, bool isDirectory = true, bool force = false)
+        {
+            if (CommonUtils.IsWindows && !CommonUtils.IsCoreApp && !path.StartsWith(@"\\?\") &&
+                (path.Length > (isDirectory ? MaxDirLength : MaxPathLength) || force))
+            {
+                if (path.StartsWith(@"\\"))
+                {
+                    return $@"\\?\UNC\{path.Remove(2)}";
+                }
+
+                path = path.NormalizeDirSeparator();
+
+                return $@"\\?\{path}";
+            }
+
+            return path.NormalizeDirSeparator();
+        }
+
+        public static string NormalizeDirSeparator(this string path)
+        {
+            string notPlatformSeparator = CommonUtils.IsWindows ? "/" : "\\";
```

Comment:
i think that `Path.AltDirectorySeparatorChar` property is better for this case, cause it contains '/' for Windows and Mac Os, and '\\' for Linux systems

Reply:
Not correct: `AltDirectorySeparatorChar` is the same as `DirectorySeparatorChar` on Linux, see https://github.com/dotnet/docs/issues/3894